File: project-1-migration/python-sample/src/stats.py
"""Stats helpers (migration sample)."""


def mean(values):
    if not values:
        raise ValueError("empty")
    return sum(values) / len(values)


def clamp(value, low, high):
    return max(low, min(high, value))


def pct_change(old, new):
    if old == 0:
        raise ValueError("old must be non-zero")
    return (new - old) / old * 100.0

File: project-1-migration/js-sample/src/stats.mjs
// MIGRATED: ESM (after migration)
export function mean(values) {
  if (values.length === 0) throw new Error("empty");
  return values.reduce((s, v) => s + v, 0) / values.length;
}
export function clamp(value, low, high) { return Math.max(low, Math.min(high, value)); }
export function pctChange(oldV, newV) {
  if (oldV === 0) throw new Error("old must be non-zero");
  return ((newV - oldV) / oldV) * 100;
}
